provisioning/utils.go

```diff
@@ -48,21 +49,30 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
 	return hostIP, err
 }
 
-func GetServerInternalIP(osclient osclientset.Interface) (string, error) {
+func GetServerInternalIP(osclient osclientset.Interface, config *metal3iov1alpha1.ProvisioningSpec, client kubernetes.Interface, targetNamespace string) (string, error) {
 	infra, err := osclient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
 	if err != nil {
 		err = fmt.Errorf("Cannot get the 'cluster' object from infrastructure API: %w", err)
 		return "", err
 	}
-
-	// NOTE(dtantsur): do we need to handle non-BareMetal platforms here?
-	if infra.Status.PlatformStatus.BareMetal == nil {
-		err = fmt.Errorf("Cannot detect server API VIP: not a baremetal platform")
+	switch infra.Status.PlatformStatus.Type {
+	case osconfigv1.BareMetalPlatformType, osconfigv1.OpenStackPlatformType, osconfigv1.VSpherePlatformType:
+		// FIXME(dtantsur): handle the new APIServerInternalIPs field and the dualstack case.
+		return infra.Status.PlatformStatus.BareMetal.APIServerInternalIP, nil
+	//NOTE(janders) how about osconfigv1.OvirtPlatformStatus, osconfigv1.KubevirtPlatformStatus, osconfigv1.NutanixPlatformStatus?
+	// they do have the APIServerInternalIP field but I suppose they don't support disabling provisioning network?
+	case osconfigv1.NonePlatformType:
+		//TODO(janders) do things the old way, reusing master code to get the patch up, further discussion needed
+		//(temporarily) re-added all the parameters needed by this code
+		if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && !config.VirtualMediaViaExternalNetwork {
+			return config.ProvisioningIP, nil
+		} else {
+			return getPodHostIP(client.CoreV1(), targetNamespace)
+		}
+	default:
```

Comment:
Let's leave a comment in https://github.com/openshift/cluster-baremetal-operator/blob/master/controllers/provisioning_enabled.go#L43 to update this code whenever that part is updated.

Reply:
Good point, done.